### DISCUSSION TITLE
How to trace syslog ingest? (Some logs not appearing in index)

### DISCUSSION THREAD

**Discussion by tomjones1977** [Security-Onion-Solutions/securityonion#5413] [Unsupported Versions]
I've been working on tweaking the syslog ingest pipeline within Elasticsearch.  Having followed the guide and edited the `syslog` config file in `/opt/so/saltstack/local/salt/elasticsearch/ingest` the index partially works - some logs are ingested as expected.

I used the Ingest Pipeline debugger to test a syslog message against the pipeline and it runs through to completion, with the correct index fields popuated, geo ip info added from the `common` pipeline etc.  However this same log does not appear anywhere in the index using the Discover tab - it has just disappeared, as have all messages with a similar format.  Other syslog messages are still being processed as expected.

The only thing I can think of is that Elastic doesn't like the fact that some fields don't currently exist in the Index Patterns.  In SO 16.04 this didn't seem to matter, and extra fields were created but not searchable until they were added to the index.  Could this be the problem I'm seeing?

More importantly, to help me answer my own questions - is there a way to debug the flow of a syslog entry so that I can see where it is failing?

**Comment by tomjones1977** [comment]
To add to this - it does appear that dynamic mapping is enabled for the index, which should auto-add new fields as they are processed.

**Reply by weslambert** [reply]
If you perform a search like the following from the CLI (from search node, or node containing the data), do you see the message?

`sudo so-elasticseearch-query $index/_search?q=<whatever you can use to filter the message>`
